src/OrdersList/OrdersDriver.cpp

```diff
@@ -179,45 +178,42 @@ void testOrdersLists() {
 
     cout << "...Printing Players orders..\n";
     cout << "-------------------------------\n";
-    for (auto o : p->getOrderList()) {
-        cout << o;
-    }
-    cout << "\n";
+    cout << p->getOrdersList();
 
     cout << "...Setting players 2nd and 4th orders to valid..\n";
-    p->getOrderList().at(1)->setValid(true);
-    p->getOrderList().at(3)->setValid(true);
+    p->getOrdersList()->getOL()->at(1)->setValid(true);
+    p->getOrdersList()->getOL()->at(3)->setValid(true);
     cout << "Players 2nd and 4th orders are valid.\n\n";
 
 
     cout << "If Player is inside execute  rders state and will be able to execute!\n\n";
     cout << "...Executing players orders...\n";
     cout << "-------------------------------\n";
-    for (auto o : p->getOrderList()) {
-        o->execute(p->getState());
-    }
-    cout << "\n";
-
+    p->getOrdersList()->executeAll(p->getState());
 
 
     cout << "\n\n---------> Test 8: Deleting Pointers <---------" << "\n\n\n";
 
     cout << "...Deleting pointers..\n";
-    OL.deleteOrdersList();
-    OLCopy.deleteOrdersList();
+    OL->deleteOrdersList();
+    delete OL;
+    OL = NULL;
+    OLCopy->deleteOrdersList();
+    delete OLCopy;
+    OLCopy = NULL;
     delete t1OL;
     t1OL = NULL;
     delete t2OL;
     t2OL = NULL;
     delete hOL;
     hOL = NULL;
-    for (auto o : p->getOrderList()) {
-        delete o;
-        o = NULL;
-    }
+    p->getOrdersList()->deleteOrdersList();
+    delete p->getOrdersList();
```

Comment:
check players MR. player is responsible for deleting orderlist, leave ur code like this, but just so u are aware, I changed it to just have `delete p` which will call the desconstructor of player which will delete the pointer to  orderlist! :D

Reply:
I can't find it, you're going to have to show me!

Reply:
nvm, I did this before the MR so i didn't see it!